src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -160,6 +158,19 @@ public inherited sharing class RD2_OpportunityEvaluationService {
         return this;
     }
 
+    private Boolean evaluteInstallmentAutoCreationSetting(npe03__Recurring_Donation__c rd) {
```

Comment:
The RD settings Installment Opp Auto Creation actions/options are evaluated for every RD in the transaction. The settings are applicable to all traversed RDs while the related Opp list can be checked outside of settings. 
The PR #4903  contains [RD2_Settings file](https://github.com/SalesforceFoundation/NPSP/blob/87b8a42bbad9811f4822e75c7ffb0a2f56eef1bc/src/classes/RD2_Settings.cls). See how the file is used with regards to new settings and see if it can be incorporated in this PR?